### ISSUE TITLE
Overloaded function (with templates as change) is not displayed in html page

### ISSUE THREAD

**Issue by zchrissirhcz** [doxygen/doxygen#8921]
**Describe the bug**
I' like to display the following 3 functions in doxygen web page, the 2nd and the 3rd are overloaded each other:
```c++
void nice();

void hello(const char* name);

template<typename T>
void hello(const char* name);
```

**Expected behavior**
The web page should display two `hello()` function.

**Screenshots**
![图片](https://user-images.githubusercontent.com/3831847/143830913-96f4000b-12ce-4501-9d88-86e8d5357162.png)

As illustrated in the screenshot, there is only one `hello()` function.

**To Reproduce**

[test.tar.gz](https://github.com/doxygen/doxygen/files/7616581/test.tar.gz)

**Version**
```
doxygen --version

1.9.3 (c83813394ceb75aff58341f70f86fad87f8c97d0)
```

**Stack trace**

**Additional context**
The C/C++ compiler I'm using is `clang version 13.0.0 (https://github.com/llvm/llvm-project/ 24c8eaec9467b2aaf70b0db33a4e4dd415139a50)
`

**Comment by zchrissirhcz** [comment]
So let's make it more suitable to debug. Based on the previously pasted C++ code, now adding `@brief` for the tempalted `hello()`, i.e.

```c++
#pragma once

void nice();

void hello(const char* name);

/// @brief the templated one
template<typename T>
void hello(const char* naming);
```

Then the generated html would absorb the templated `hello()`'s brief message to the non-templated one:

![图片](https://user-images.githubusercontent.com/3831847/144428702-c61bf622-b78f-4430-9cd1-8ced0d09103f.png)

**Comment by zchrissirhcz** [comment]
Now turn on the debug print for functions:
```c++
//Debug::DebugMask Debug::curMask = Debug::Quiet;
Debug::DebugMask Debug::curMask = Debug::Functions;
```

And get:
```
FUNCTION_SEC:
  'void' ''::'nice' '()' relates='' relatesType='0' file='/home/zz/work/test/doxygen_example3/src/hello.h' line='3' bodyLine='-1' #tArgLists=0 mGrpId=-1 spec=0 proto=1 docFile=
  --> new function nice found!
  Global Function:
    'void' ''::'nice' '()' proto=1
    def='void nice'
FUNCTION_SEC:
  'void' ''::'hello' '(const char *name)' relates='' relatesType='0' file='/home/zz/work/test/doxygen_example3/src/hello.h' line='5' bodyLine='-1' #tArgLists=0 mGrpId=-1 spec=0 proto=1 docFile=
  --> new function hello found!
  Global Function:
    'void' ''::'hello' '(const char *name)' proto=1
    def='void hello'
FUNCTION_SEC:
  'void' ''::'hello' '(const char *naming)' relates='' relatesType='0' file='/home/zz/work/test/doxygen_example3/src/hello.h' line='9' bodyLine='-1' #tArgLists=1 mGrpId=-1 spec=0 proto=1 docFile=/home/zz/work/test/doxygen_example3/src/hello.h
  --> function hello already found!
()<->()
(const char *name)<->(const char *name)
(const char *name)<->(const char *naming)
/home/zz/work/test/doxygen_example3/src/hello.h:9: warning: no matching file member found for 
void hello(const char *naming)
Possible candidates:
 'void hello(const char *name)' at line 5 of file /home/zz/work/test/doxygen_example3/src/hello.h
```

Among which, `  --> function hello already found!` is very-not-OK.

**Comment by zchrissirhcz** [comment]
Thus, if I just change `doxygen.cpp`, line 3588, from previous:
```c++
if ((mn=Doxygen::functionNameLinkedMap->find(rname)))
```
To this:
```c++
if (0 && (mn=Doxygen::functionNameLinkedMap->find(rname)))
```

Then I can correctly get what I expected, for the specific example I provided:

![图片](https://user-images.githubusercontent.com/3831847/144430814-03550fa6-0a4d-4c57-8b7c-c9428ab5683f.png)

**Comment by doxygen** [comment]
@zchrissirhcz I've made a slightly different fix than your pull request (https://github.com/doxygen/doxygen/pull/8930/commits/d80f4e4a0ee4cf57d5d617d30a88cc49aa26c4cd). Please verify if it also fixes the problem.

**Comment by zchrissirhcz** [comment]
@doxygen Thanks for this fix. I compiled the lastest commit of doxygen, it does solve the template and non-templated function conflict.

( Btw, document coverage with coverxygen still failed, don't know if there is a better solution than https://github.com/doxygen/doxygen/pull/8937 for https://github.com/doxygen/doxygen/pull/331 )

**Comment by albert-github** [comment]
Please don't close the issue as this will be done at the moment of an official release

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.3.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
